src/server/game/Server/Packets/ItemPackets.cpp

```diff
@@ -544,3 +553,11 @@ WorldPacket const* WorldPackets::Item::SocketGemsResult::Write()
 
     return &_worldPacket;
 }
+
+WorldPacket const* WorldPackets::Item::ItemUpgradeResult::Write()
+{
+    _worldPacket.WriteBit(Success);
+    _worldPacket.FlushBits();//dose it needed?
```

Comment:
to answer this question - yes, you must call FlushBits if writing a bit is the last thing done in a packet (and remove the comment since I just answered it)